ext/drx/drx.c

```diff
@@ -2569,30 +2581,33 @@ drx_try_to_detect_scatter_gather_sequence(
     params.scalar_mask_update_no = 0;
     params.info = info;
     params.sg_info = sg_info;
-    params.inst = inst;
     params.pc = params.info->fragment_info.cache_start_pc;
+    instr_init(drcontext, &params.inst);
     /* As the state machine is looking for blocks of code that the fault may hit, the 128
      * bytes is a conservative approximation of the block's size, see (a) and (b) above.
      */
     while (params.pc <= params.info->raw_mcontext->pc + 128) {
-        instr_reset(drcontext, params.inst);
+        instr_reset(drcontext, &params.inst);
         params.prev_pc = params.pc;
-        params.pc = decode(drcontext, params.pc, params.inst);
+        params.pc = decode(drcontext, params.pc, &params.inst);
         if (params.pc == NULL) {
             /* Upon a decoding error we simply give up. */
-            return true;
+            goto drx_restore_state_scatter_gather_exit;
```

Comment:
These gotos could all just be breaks, right?

Reply:
Done.